packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -0,0 +1,113 @@
+import React, { PropTypes } from 'react';
+import moment from 'moment';
+import TextMask from 'react-text-mask';
+import TimePipe from './TimePipe';
+import './TimeInput.scss';
+
+const propTypes = {
+  /**
+   * A moment object to use as the default initial time.
+   */
+  defaultTime: PropTypes.object,
+    /**
+   * A callback function to execute when a time value is entered.
+   */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  defaultTime: null,
+  onChange: null,
+};
+
+class TimeInput extends React.Component {
+  static userLocale() {
+    // TODO: Get the locale from i18nProvider
+    return 'en-US';
+  }
+
+  static timeFormat() {
+    // TODO: Get the time format translation.
+    return 'HH:mm';
+  }
+
+  static formattedTime(timeMoment) {
+    if (timeMoment) {
+      return timeMoment.clone().locale(TimeInput.userLocale()).format(TimeInput.timeFormat());
+    }
+
+    return '';
```

Comment:
If no default time is passed in the prop or there is no value in the input (the user manually backspaced to delete the time), then it would display the placeholder text (i.e.; HH:mm) in the input.